### PR TITLE
Update component models

### PR DESCRIPTION
**Pull request by mind-adobe**
Add component definition / models / filters for Search, Video and Table block
For the table block I added variations that correspond to the number of columns 1-5 for each row entry.

Test URLs:
- Before: https://main--aem-block-collection-xwalk--adobe-rnd.hlx.page
- After: https://update-component-models--aem-block-collection-xwalk--adobe-rnd.hlx.page

### CODE REVIEW

File: blocks/table/table.js [hunk 1]
```diff
@@ -4,33 +4,32 @@
  * https://www.hlx.live/developer/block-collection/table
  */
```
Comment (msagolj) [inline blocks/table/table.js:6]: - add color definition for striped class in table
- no-header for table is not working anymore
- make 2 column width 50/50

File: component-models.json [hunk 2]
```diff
@@ -214,5 +214,130 @@
...
+              }
+            ]
+          }
+        ]
+      },
+      {
+        "component": "reference",
+        "valueType": "string",
+        "name": "placeholder_image",
+        "label": "Image",
```
Comment (msagolj) [inline component-models.json]: use more descriptive label

**Review by msagolj — CHANGES_REQUESTED** [review]

**Review by msagolj — APPROVED** [review]

File: component-models.json [hunk 3]
```diff
@@ -214,5 +214,251 @@
...
+            "name": "Header",
+            "children": [
+              {
+                "name": "No Header",
+                "value": "no-header"
+              }
+            ]
+          },
+          {
+            "name":"Columns",
```
Comment (buuhuu) [inline component-models.json]: Why not create a new select for this to change the value of the `filter` property to anything of `table`, `table-3-columns`, `table-4-columns`, ... etc.

File: component-filters.json [hunk 4]
```diff
@@ -42,5 +45,35 @@
...
+  {
+    "id":"search",
+    "components":[]
+  },
+  {
+    "id":"video",
+    "components":[]
+  },
+  {
+    "id":"table-2-rows",
```
Comment (buuhuu) [inline component-filters.json]: probably better call it `table-2-columns`?

File: scripts/editor-support.js [hunk 5]
```diff
@@ -10,6 +10,19 @@ import {
 import { decorateRichtext } from './editor-support-rte.js';
 import { decorateMain } from './scripts.js';
 
+/**
+ *
+ * @param {Element} block
+ */
+function updateTableFilter(block) {
```
Comment (buuhuu) [inline scripts/editor-support.js]: not needed if you change the filter directly with the table's component model

**Pull request merged**